server/src/api/modules/groups/services/FetchGroupStatisticsService.ts

```diff
@@ -59,8 +61,10 @@ async function fetchGroupStatistics(payload: FetchGroupStatisticsParams): Promis
   });
 
   const averageStats = format(averageSnapshot, averageEfficiencyMap);
+  const players = await playerServices.findPlayers({ ids: snapshots.map(s => s.playerId) });
```

Comment:
I think it would be more efficient to

- calculate best values from everyone's snapshots
- fetch only the players that are included in the "best" list
- "insert" them into their places in the best object

In a group of 700 members, if only 50 take up the top spots, no point in fetching the other 650

Reply:
Thats very true, and here I thought I was being clever just fetching the players once XD

Reply:
Resolved in latest push.